### PR TITLE
Add a Converter Job which converts v1beta1 run in db to v1 run

### PR DESCRIPTION
**Pull request by khrm**
This ensures that only v1 records are existing in the db instead of a mix of v1beta1 and v1

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Converter to convert older records from v1beta1 Tekton object to v1 Tekton object. This is controlled by the following fields in API:
 CONVERTER_ENABLE: Whether to start converter or v1beta1 TaskRun/PipelineRun records to v1                                                                                                                                                                                               
CONVERTER_DB_LIMIT:     How many records to convert at a time in a transaction                                                                                                     |
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind feature

**Comment by khrm** [comment]
I have removed the cronjob and added a flag to the API server to enable converter. 
https://github.com/tektoncd/results/compare/e70451083446c26c8eb0a8e98b1dda5310b126ae..2f931e13096492c7b4755c22875a6400b0239492

**Comment by khrm** [comment]
Strange. There's something weird going on. Somehow it's showing changes from main branch also.

**Review by khrm — COMMENTED** [review]
/assign @vdemeester 
Please look into this.

**Comment by khrm** [comment]
@avinal Please look into this.

/cc @enarha @vdemeester

**Review by enarha — COMMENTED** [review]
/lgtm

**Comment by khrm** [comment]
@vdemeester @enarha Can you please look into this PR?

**Comment by khrm** [comment]
@enarha LGTM got removed due to rebase and error fix. Please review again.

File: cmd/api/README.md [hunk 1]
```diff
@@ -38,6 +38,8 @@
 | S3_MULTI_PART_SIZE       | S3 Multi part size                                                                                                                | 5242880 (default)                                                                            |
 | GCS_BUCKET_NAME          | GCS Bucket Name                                                                                                                   | <GCS Bucket Name>                                                                            |
 | STORAGE_EMULATOR_HOST    | GCS Storage Emulator Server                                                                                                       | http://localhost:9004                                                                        |
+| CONVERTER_ENABLE         | Whether to start converter or v1beta1 TaskRun/PipelineRun records to v1                                                           | true                                                                                         |
```
Comment (enarha) [inline cmd/api/README.md:41]:
```suggestion
| CONVERTER_ENABLE         | Whether to start conversion of v1beta1 TaskRun/PipelineRun records to v1                                                           | true                                                                                         |
```

Comment (enarha) [reply cmd/api/README.md:41]: Sorry if I missed that one on the previous review.

**Pull request merged**